clickhouse_connect/driver/buffer.py

```diff
@@ -0,0 +1,119 @@
+import sys
+import array
+
+from clickhouse_connect.driver.types import ByteSource
+
+must_swap = sys.byteorder == 'big'
+
+
+class ResponseBuffer(ByteSource):
+    slots = 'slice_sz', 'buf_loc', 'end', 'gen', 'buffer', 'slice'
+
+    def __init__(self, source):
+        self.slice_sz = 4096
+        self.buf_loc = 0
+        self.end = 0
```

Comment:
it's a buffer size?

Reply:
I like that name better -- originally it wasn't actually a buffer size, but the end of the "good" data in a fixed buffer, but the implementation changed.